dashboard/test/controllers/scripts_controller_test.rb

```diff
@@ -175,15 +180,43 @@ class ScriptsControllerTest < ActionController::TestCase
     assert_response :ok
   end
 
-  test "show: redirect to latest stable version in family for student" do
-    sign_in create(:student)
+  test "show: do not redirect when showing latest stable version in family for student" do
+    sign_in @no_progress_or_assignment_student
+    get :show, params: {id: @coursez_2018.name}
+    assert_response :success
+  end
+
+  test "show: redirect from older version to latest stable version in family for student" do
+    sign_in @no_progress_or_assignment_student
     get :show, params: {id: @coursez_2017.name}
     assert_redirected_to "/s/#{@coursez_2018.name}?redirect_warning=true"
   end
 
+  test "show: redirect from older version to latest stable version in family for logged out user" do
+    get :show, params: {id: @coursez_2017.name}
+    assert_redirected_to "/s/#{@coursez_2018.name}?redirect_warning=true"
+  end
+
+  test "show: redirect from new unstable version to latest stable version in family for student" do
+    sign_in @no_progress_or_assignment_student
+    get :show, params: {id: @coursez_2019.name}
+    assert_redirected_to "/s/#{@coursez_2018.name}?redirect_warning=true"
+  end
+
+  test "show: redirect from new unstable version to to latest stable version in family for logged out user" do
+    get :show, params: {id: @coursez_2019.name}
+    assert_redirected_to "/s/#{@coursez_2018.name}?redirect_warning=true"
+  end
+
+  test "show: redirect from new unstable version to assigned version for student" do
+    sign_in @student_coursez_2017
+    get :show, params: {id: @coursez_2019.name}
+    assert_redirected_to "/s/#{@coursez_2017.name}?redirect_warning=true"
+  end
+
   test "show: do not redirect student to latest stable version in family if they can view the script version" do
     Script.any_instance.stubs(:can_view_version?).returns(true)
```

Comment:
this one is a bit hacky. I was thinking there should be a "do not redirect to latest stable version if student goes the the version they are assigned to". not sure if this case covers that, but perhaps this case should be changed to test that (and possibly other) scenarios more explicitly? If that would be an explosion of test cases, it would help to at least have a comment here describing some of the scenarios this test case is covering.

Reply:
new comment looks good! 👍